packages/redux/src/raffles/selectors.ts

```diff
@@ -0,0 +1,89 @@
+import { getEntities, getEntityById } from '../entities/selectors';
+import {
+  getError as getErrorFromReducer,
+  getIsLoading as getIsLoadingFromReducer,
+} from './reducer';
+import type {
+  Raffle,
+  RaffleParticipation,
+} from '@farfetch/blackout-client/src';
+import type { RafflesState } from './types';
+import type { StoreState } from '../types/storeState.types';
+
+/**
+ * Returns all raffles from state.
+ *
+ * @param state - Application state.
+ *
+ * @returns Object with key values pairs representing raffleId and raffle properties.
+ */
+export const getRaffles = (state: StoreState) => getEntities(state, 'raffles');
+
+/**
+ * Returns a specific raffle by its id.
+ *
+ * @param state   - Application state.
+ * @param raffleId - raffle id.
+ *
+ * @returns Raffle normalized.
+ */
+export const getRaffle = (state: StoreState, raffleId: Raffle['id']) =>
+  getEntityById(state, 'raffles', raffleId);
+
+/**
+ * Returns the estimation for a raffle
+ * @param state - Application state
+ * @param raffleId - Raffle id.
+ *
+ * @returns Returns the object estimation raffle if exists.
+ */
+export const getRaffleEstimation = (
+  state: StoreState,
+  raffleId: Raffle['id'],
+) => {
+  return getRaffle(state, raffleId)?.estimation;
+};
+
+/**
+ * Returns the participations for a raffle.
+ * @param state - Application state
+ * @param raffleId - Raffle id.
+ * @returns Returns an array of participations ids.
+ */
+export const getRaffleParticipationsIds = (
+  state: StoreState,
+  raffleId: Raffle['id'],
+) => {
+  return getRaffle(state, raffleId)?.participations;
+};
+
+/**
+ * Returns a raffle participation by raffle participation id.
+ * @param state - Application state
+ * @param raffleId - Raffle id.
+ * @returns Object of raffle participation
+ */
+export const getRaffleParticipationById = (
+  state: StoreState,
+  raffleParticipationId: RaffleParticipation['id'],
+) => getEntityById(state, 'raffleParticipations', raffleParticipationId);
+
+/**
+ * Return loading raffles status from state.
+ *
+ * @param state - Application state.
+ *
+ * @returns boolean
+ */
+export const getIsLoading = (state: StoreState) =>
+  getIsLoadingFromReducer(state as RafflesState);
```

Comment:
Not sure about this, but you receive an argument of type `StoreState` and use the very same argument but as `RafflesState`; is this correct? 🧐

Reply:
can you see again pls ?